### PR TITLE
Borders and ticks

### PR DESCRIPTION
**Pull request by aclements**
Clean up rendering of borders and ticks.

@quentinmit

---
This change is [Reviewable](https://reviewable.io/reviews/aclements/go-gg/5)

### CODE REVIEW

File: gg/render.go [hunk 1]
```diff
@@ -259,7 +259,9 @@ func (e *eltSubplot) render(r *eltRender) {
 	// TODO: Theme.
 
 	// Render border.
-	svg.Path(fmt.Sprintf("M%d %dV%dH%d", xi+1, yi, y2i-1, x2i), "stroke:#888; fill:none; stroke-width:2") // TODO: Theme.
+	if false {
+		svg.Path(fmt.Sprintf("M%d %dV%dH%d", xi+1, yi, y2i-1, x2i), "stroke:#888; fill:none; stroke-width:2") // TODO: Theme.
```
Comment (quentinmit) [inline gg/render.go]: I definitely prefer the look with the border, FWIW.

File: gg/render.go [hunk 2]
```diff
@@ -325,14 +326,14 @@ func renderScale(svg *svg.SVG, dir rune, scale Scaler, ticks plotEltTicks, pos i
...
+				fmt.Fprintf(&path, "M%.6g %dv%.6g", p, pos, -t.length)
 			} else {
-				path = append(path, fmt.Sprintf("M%d %.6gh%.6g", pos, p, t.length))
+				fmt.Fprintf(&path, "M%d %.6gh%.6g", pos, p, t.length)
 			}
 		}
 
-		svg.Path(strings.Join(path, ""), "stroke:#888; stroke-width:2") // TODO: Theme
 	}
+	svg.Path(path.String(), "stroke:#888; stroke-width:2") // TODO: Theme
```
Comment (quentinmit) [inline gg/render.go]: The SVG spec recommends adding newlines every <255 characters (but the newline must be at the boundary of a command); are we worried that we'll now produce paths that are too long? (Of course, the paths for the plot itself are probably even worse...)

**Comment by quentinmit** [comment]
Are you sure that even with rounding this will be drawn correctly? e.g. the clipping path won't cut off part of the antialiasing?

File: gg/render.go (hunk 1 above)
Comment (aclements) [inline gg/render.go]: Okay. I'll give it another chance and see what I think.

File: gg/render.go (hunk 2 above)
Comment (aclements) [inline gg/render.go]: Interesting. It probably would be good to wrap them, but I'll do that as a separate commit.

**Comment by aclements** [comment]
> Are you sure that even with rounding this will be drawn correctly?

I've moved the border and ticks outside the clip path.

**Comment by quentinmit** [comment]
Your wrapping code scares me but it appears to be correct. LGTM

**Comment by aclements** [comment]
> Your wrapping code scares me but it appears to be correct. LGTM

Whoops. Except that I left it at the "wrap at 20 characters" testing setting.

Any suggestions for how to make it less "scary"?

**Pull request merged**